src/App.tsx

```diff
@@ -17,18 +44,33 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                selectedCategory={selectedCategory}
+                setSelectedCategory={setSelectedCategory}
+                inputValue={inputValue}
+                setInputValue={setInputValue}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading && <Loader />}
+
+              {selectedTodo && (
+                <TodoModal
+                  selectedTodo={selectedTodo}
+                  setSelectedTodo={setSelectedTodo}
+                />
+              )}
+
+              <TodoList
+                todos={filteredTodos}
+                selectedTodo={selectedTodo}
+                setSelectedTodo={setSelectedTodo}
```

Comment:
The `setSelectedTodo` function should allow setting the state to `null` to close the modal, as required by the task: 'x button should close the modal'. Ensure that the `TodoList` component's `setSelectedTodo` prop can handle `null` values.